### DISCUSSION TITLE
Issue when i call to bitcoin_testnet rpc

### DISCUSSION THREAD

**Discussion by ghost** [btcsuite/btcd#2161] [Q&A]
![Screenshot from 2024-04-08 11-27-22](https://github.com/btcsuite/btcd/assets/146687333/e8f1b0cc-5ff5-4c3a-80a6-c778229e5c7c)
Make sure that the user and password are correct:

let rpc_url = "http://172.14.198.19:18444";
  let rpc = Client::new(rpc_url, Auth::UserPass("bitcoin_regtest".to_string(), "1".to_string()))
      .unwrap();
  let block_height = rpc.get_block_count().unwrap();
  println!("Current block height: {}", block_height);

**Comment by guggero** [comment]
Wrong repo? `btcd` is written in Go, your code seems to be Rust. Or are you trying to connect to `btcd` from Rust?
More details please.

**Comment by ghost** [comment]
@guggero  yes, i'm trying to connect to btcd from Rust, so it's possible?

**Comment by guggero** [comment]
Yes, it should be possible. The error seems to be "Connection reset by peer". So you're using the wrong port or `btcd` isn't ready.

**Comment by ghost** [comment]
@guggero hmm, i don't think so, btcd ready and I have tried all possible ports
![Screenshot from 2024-04-08 15-47-35](https://github.com/btcsuite/btcd/assets/146687333/3e607125-c4c5-43da-9e63-6954d6898645)
![Screenshot from 2024-04-08 15-51-42](https://github.com/btcsuite/btcd/assets/146687333/69a46fb6-df3b-49ec-8067-dfcb8a11e02a)

**Comment by guggero** [comment]
Then it's probably an issue with the IP address or firewall that you can't reach `btcd` from the Rust application.
